### ISSUE TITLE
feat(rules): enforce li role constraint based on parent role

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#3636]
## Summary

Add validation for `<li>` element role restrictions that depend on the parent element's role.

## What to enforce

- `<li>` descendant of `role=listbox` or `role=list` → role must be `option` only
- `<li>` descendant of `role=menu` or `role=menubar` → role must be one of `menuitem`, `menuitemcheckbox`, `menuitemradio`
- `<li>` descendant of `role=tablist` → role must be `tab` only
- `<li>` descendant of `role=tree` → role must be `treeitem` only
- `<li>` descendant of `<ul>`, `<ol>`, `<menu>` with no explicit role → role must be `listitem`, `menuitem`, `menuitemcheckbox`, `menuitemradio`, `option`, `none`, `presentation`, `radio`, `separator`, `tab`, `treeitem`

## Spec reference

- https://w3c.github.io/html-aria/#el-li

## Impact

~6 missed errors in the nu-html-checker compatibility benchmark.

**Comment by YusukeHirao** [comment]
## nu-html-checker 実出力での検証

Docker 経由で nu-html-checker を実行して確認:

### テスト HTML
```html
<ul role="listbox"><li role="menuitem">x</li></ul>
<ul role="menu"><li role="option">x</li></ul>
<ul><li role="button">x</li></ul>
<ul role="tablist"><li role="treeitem">x</li></ul>
```

### nu-html-checker 出力
7 件の error（li role 制約 + parent/owned 制約）

### markuplint 出力
**7 件の problems（4 errors + 3 warnings）を既に検出**:
- `wai-aria-permitted-roles` → `li role=button` を検出
- `wai-aria-required-owned-elements` → listbox/menu/tablist の子ロール不整合を検出
- `wai-aria-required-parent-role` → menuitem/option/treeitem の親ロール不整合を検出

### 判定: OVERSTATED

実体はカバー済み。個別ルール（`wai-aria-required-parent-role`, `wai-aria-required-owned-elements`, `wai-aria-permitted-roles`）の組み合わせで既に検出されている。メッセージが nu-validator とは異なるが、検出漏れではない。

li 固有の新ルール追加はメッセージが cleaner になる程度の価値で、必要性が低い。

→ **クローズ推奨**

**Comment by YusukeHirao** [comment]
Closing as overstated. Verification showed existing rules (wai-aria-permitted-roles + wai-aria-required-parent-role + wai-aria-required-owned-elements) already cover all cases. See https://github.com/markuplint/markuplint/issues/3636#issuecomment-4248529577 for details.
